### ISSUE TITLE
Missing Release for 0.9.0 on Github

### ISSUE THREAD

**Issue by zetamorph** [argoproj-labs/argocd-operator#1295]
Hi,

it seems that there is a new version, 0.9.0, of the ArgoCD operator on OperatorHub while Github is still showing the 0.8.0 release as the latest one. 

**Screenshots**

![Bildschirmfoto 2024-04-03 um 14 40 26](https://github.com/argoproj-labs/argocd-operator/assets/23425025/0f44af63-d085-476b-92ee-e0404b9cfd71)

**Comment by bo0ts** [comment]
Can confirm. Looking at https://quay.io/repository/argoprojlabs/argocd-operator?tab=history&tag=latest v0.9.0 was overwritten 
```
v0.9.0 was moved to SHA256[ 25c71c9f0fbc ](https://quay.io/repository/argoprojlabs/argocd-operator/manifest/sha256:25c71c9f0fbc5203b3ea8021cb770a98642ddf3ee3cc85faac3be642b7183491) from SHA256 38d61c3acda6
```

The old manifest must have been deleted: https://quay.io/repository/argoprojlabs/argocd-operator/manifest/sha256:38d61c3acda6230525614d1e609d0955f944680ccf12c07c1c84a7e5b0b98de8

**Comment by peterroth** [comment]
Unfortunately, we had automatic update enabled and the process failed. The service is up and running, but 0.8.0 stuck in "Replacing" status while 0.9.0 cannot start up and logs an error message continuously:
```Generated from operator-lifecycle-manager
9 times in the last 0 minutes
install strategy failed: Deployment.apps "argocd-operator-controller-manager" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"control-plane":"argocd-operator"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable```

I was able to reproduce it just now, running ArgoCD on a Single Node OKD environment. This might be the reason why the new version is deleted on operatorhub.io.

**Comment by Elyytscha** [comment]
had the same problem, my okd auto upgraded to 0.9.0 and i had a 'jiatan75' moment where I thought about whats going on, on operatorhub its still 0.8.0, normally operatorhub updates faster then the redhat community operator catalog, so i investigated.

this commit should have fixed it https://github.com/argoproj-labs/argocd-operator/pull/1296/files#diff-01184a8cb2250baf03253df4c205a04a94228e5db47e1cbfd7e9edf44182a65a 

but https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/4321#issuecomment-2035333231 said nop because of
> The PR modifies existing bundles: ['argocd-operator/0.9.0']
(https://gist.github.com/rh-operator-bundle-bot/86fa0757b79b22cffbfd9003c7b97bc8)

so I think what needs to be done is to release a 0.9.1 csv over here in the argocd operator repo
then update or create a new pr with 0.9.1 in community-operators-prod repo

**Comment by bo0ts** [comment]
@reginapizza Any chance for an update soon?

**Comment by stanbog** [comment]
Hello,

Do we have an update on this ?

Thank you.

**Comment by bo0ts** [comment]
@reginapizza Looks like the PR is approved but not merged yet?

**Comment by reginapizza** [comment]
@bo0ts hey! Yeah unfortunately the issue with pre-flight is not fixed yet. If you see most of the other open PR's the majority of them are also failing the same way. I've been keeping up with the people working on fixing it on slack and waiting for updates on when it should be fixed, and in the meantime continuing to try the pipelines in case there's some chance that it could pass :/ I'm sorry that it's not fixed yet, I'm trying everything I can to get it merged asap and I'll post here as soon as I see it's merged!

**Comment by GeroL** [comment]
The image vor 0.9.1 was pushed successfully:
```
[build-bundle : build] Successfully tagged quay.io/community-operator-pipeline-prod/argocd-operator:0.9.1-5cff466ceb3d7977e87f0cae0c751f219c6f4b40
[build-bundle : build] 794fdd3b2e06703c3d02727c008940ad936d0751fc8cf19d8c893788e22433c0
[build-bundle : build] Pushing quay.io/community-operator-pipeline-prod/argocd-operator:0.9.1-5cff466ceb3d7977e87f0cae0c751f219c6f4b40
[build-bundle : build] Getting image source signatures
[build-bundle : build] Copying blob sha256:5c3fd3a19596468dda16f64b15bd4650f4938489aa6c63d5dcf51aa888d03e27
[build-bundle : build] Copying config sha256:794fdd3b2e06703c3d02727c008940ad936d0751fc8cf19d8c893788e22433c0
[build-bundle : build] Writing manifest to image destination
[build-bundle : build] Storing signatures
[build-bundle : build] sha256:8848f3189af43a46c2167488f485e2f589585ce8ca0364379d116a5d73caf9ba
```

**Comment by reginapizza** [comment]
yep! Just saw that the PR was finally merged, so 0.9.1 should be available now. If you find it acceptable, please mark the issue as closed!

**Comment by stanbog** [comment]
Hello,

Can confirm that the issue is fixed for us as well.

Thank you for all the efforts !

**Comment by jonasbartho** [comment]
hmm, in my OKD4-staging environment our ArgoCD operator is not able to see any newer versions than v0.8.0

We noticed in the ArgoCD-jobs under namespace `openshift-marketplace` that it wants to pull from `quay.io/openshift-community-operators/argocd-operator@sha256:0...`... while it actually should pull from `quay.io/community-operator-pipeline-prod/argocd-operator@sha256:6`  ....

Any tips here? I have tried reinstalling the operator(with first removing the sub,csv and relevant job/cm under namespace `openshift-marketplace`) without any luck.. :)

**Comment by jonasbartho** [comment]
> Hi @reginapizza  
> 
> hmm, in my OKD4-staging environment our ArgoCD operator is not able to see any newer versions than v0.8.0
> 
> We noticed in the ArgoCD-jobs under namespace `openshift-marketplace` that it wants to pull from `quay.io/openshift-community-operators/argocd-operator@sha256:0...`... while it actually should pull from `quay.io/community-operator-pipeline-prod/argocd-operator@sha256:6` ....
> 
> Any tips here? I have tried reinstalling the operator(with first removing the sub,csv and relevant job/cm under namespace `openshift-marketplace`) without any luck.. :)

This was due to a bug in OKD4. Pruning alle the community-operator images with crictl resolved the issue.

**Comment by jonasbartho** [comment]
> peterroth

Hi Peter, we are hitting the same bug. Can you share what you did to fix it?

**Comment by jwklijnsma** [comment]
is there any solution to fix this with reinstall the operator enzo ?
